src/goose.rs

```diff
@@ -1695,29 +1753,26 @@ impl GooseUser {
     /// task!(setup_custom_client).set_on_start();
     ///
     /// async fn setup_custom_client(user: &GooseUser) -> GooseTaskResult {
-    ///   use reqwest::{Client, header};
+    ///     use reqwest::{Client, header};
     ///
-    ///   // Build a custom HeaderMap to include with all requests made by this client.
-    ///   let mut headers = header::HeaderMap::new();
-    ///   headers.insert("X-Custom-Header", header::HeaderValue::from_str("custom value").unwrap());
+    ///     // Build a custom HeaderMap to include with all requests made by this client.
+    ///     let mut headers = header::HeaderMap::new();
+    ///     headers.insert("X-Custom-Header", header::HeaderValue::from_str("custom value").unwrap());
     ///
-    ///   let builder = Client::builder()
-    ///     .default_headers(headers)
-    ///     .user_agent("custom user agent")
-    ///     .cookie_store(true);
+    ///     let builder = Client::builder()
+    ///         .default_headers(headers)
+    ///         .user_agent("custom user agent")
+    ///         .cookie_store(true);
     ///
-    ///   user.set_client_builder(builder);
-    ///   Ok(())
+    ///     user.set_client_builder(builder).await?;
+    ///
+    ///     Ok(())
     /// }
     /// ```
-    pub async fn set_client_builder(&self, builder: ClientBuilder) {
-        match builder.build() {
-            Ok(c) => *self.client.lock().await = c,
-            Err(e) => {
-                error!("failed to build web client: {}", e);
-                std::process::exit(1);
-            }
-        };
+    pub async fn set_client_builder(&self, builder: ClientBuilder) -> Result<(), GooseTaskError> {
+        *self.client.lock().await = builder.build()?;
```

Comment:
Out of interest: Is *self == self.unwrap()?

Reply:
We pass in a reference (&self). The `*` dereferences this memory pointer it so we can grab the actual lock and set the builder contained.